client/session.go

```diff
@@ -1693,6 +1852,47 @@ func (s *session) streamAndGroupCollectedBlocksMetadata(
 	}
 }
 
+func (s *session) streamAndGroupCollectedBlocksMetadataV2(
+	peersLen int,
+	metadataCh <-chan blocksMetadata,
+	enqueueCh *enqueueChannel,
+) {
+	metadata := make(map[hashAndBlockStart]*receivedBlocks)
```

Comment:
Maybe we should make this `map[hashAndBlockStart]receivedBlocks` to avoid heap alloc per entry and just update it in place?

Reply:
I thought about changing this, but the existing code uses a pointer. I think this codepath is hot enough that I'm concerned about changing it while we're also trying to measure lots of other perf-related stuff.

Reply:
Maybe a TODO or task to investigate would be sufficient?

Reply:
spoke in person, decided to leave this in for now to not pollute our perf measurements